api/api_ethereum.go

```diff
@@ -792,15 +793,19 @@ func newEthRPCTransaction(tx *types.Transaction, blockHash common.Hash, blockNum
 		result.GasTipCap = (*hexutil.Big)(tx.GasTipCap())
 		// TODO-Klaytn: If we change the gas price policy from fixed to dynamic,
 		// we should change the params.ZeroBaseFee(fixed value) to dynamic value.
-		price := math.BigMin(new(big.Int).Add(tx.GasTipCap(), new(big.Int).SetUint64(params.ZeroBaseFee)), tx.GasFeeCap())
-		result.GasPrice = (*hexutil.Big)(price)
+		if b != nil && b.Header().BaseFee != nil {
+			result.GasPrice = (*hexutil.Big)(b.Header().BaseFee)
+		} else {
+			price := math.BigMin(new(big.Int).Add(tx.GasTipCap(), new(big.Int).SetUint64(params.ZeroBaseFee)), tx.GasFeeCap())
```

Comment:
It comes from `EffectiveGasPrice( )` in ethereum. As I know Klaytn have used just for returning `unitPrice` in case `GasTipCap( ) == GasFeeCap( ) == unitPrice and BaseFee == 0`.

Reply:
What relationship is there between returning `unitPrice` and adding zero value?

Reply:
Nothing, It has been used for returning unitPrice as logically same

Reply:
Why is it then placed still even that code does not affect? This was my original question.

Reply:
I don't know. In the past, I thought klaytn might go with a dynamic price, so the developers left possible related codes at that time. This is the conclusion I have found so far. I want you to stop discussing this.

Reply:
You have a misapprehension. I didn't ask for a history of this (`Why is it then placed still even that code does not affect? `). Now that the dynamic pricing is adopted and the new parameter (`ZeroBaseFee`) is newly changed (actually renamed), the code might have been gotten rid of it this time if that does not affect still.